proxy/src/http/conn_pool.rs

```diff
@@ -220,54 +219,72 @@ async fn connect_to_compute(
 
     let node_info = &mut creds.wake_compute(&extra).await?.expect("msg");
 
-    // This code is a copy of `connect_to_compute` from `src/proxy.rs` with
-    // the difference that it uses `tokio_postgres` for the connection.
     let mut num_retries = 0;
-    let mut should_wake = true;
+    let mut wait_duration = time::Duration::ZERO;
+    let mut should_wake_with_error = None;
     loop {
+        if !wait_duration.is_zero() {
+            time::sleep(wait_duration).await;
+        }
+
+        // try wake the compute node if we have determined it's sensible to do so
+        if let Some(err) = should_wake_with_error.take() {
+            match try_wake(node_info, &extra, &creds).await {
+                // we can't wake up the compute node
+                Ok(None) => return Err(err),
+                // there was an error communicating with the control plane
+                Err(e) => return Err(e.into()),
+                // failed to wake up but we can continue to retry
+                Ok(Some(ControlFlow::Continue(()))) => {
+                    wait_duration = retry_after(num_retries);
+                    should_wake_with_error = Some(err);
+
+                    num_retries += 1;
```

Comment:
This code is a bit hard to read, but if I understand it correctly, we use the same var `num_retries` for both connection to compute retries and control-plane request (wake compute) retries. At the same time, const is named `NUM_RETRIES_WAKE_COMPUTE`

And even though `NUM_RETRIES_WAKE_COMPUTE` set to 1, considering that we first use a cached address for connection:
- if connection connection to compute will fail, we will have `num_retries = 0`
- if then we do wake request and fail, we will have `num_retries = 1`
- if next time wake succeeds, we will try to connect again

So it feels like `NUM_RETRIES_WAKE_COMPUTE` is rather applies to `connect_to_compute_once`, as it doesn't limit wake requests count

It could be I just read it wrongly, though, but I'd make it clear that we have one global const limit and it applied to both wake retries and compute connection retries

Reply:
> And even though NUM_RETRIES_WAKE_COMPUTE set to 1

In my previous PR, I increased this to 10